src/main/groovy/com/sequenceiq/cloudbreak/client/CloudbreakClient.groovy

```diff
@@ -130,6 +133,19 @@ class CloudbreakClient {
         return response?.data?.id
     }
 
+    def String postRecipe(String recipe, Boolean publicInAccount) throws Exception {
+        log.debug("Posting recipe to account...")
+        def binding = ["RECIPE": recipe]
+        def response;
+        if (publicInAccount) {
+            response = processPost(Resource.ACCOUNT_RECIPES, binding)
+        } else {
+            response = processPost(Resource.USER_RECIPES, binding)
+        }
+        log.debug("Got response: {}", response.data.id)
+        return response?.data?.id
```

Comment:
I can change it but all the other post methods return String for some
strange reason :)
On 2 Feb 2015 20:08, "Krisztian Horvath" <notifications@github.com> wrote:

> In
> src/main/groovy/com/sequenceiq/cloudbreak/client/CloudbreakClient.groovy
> <https://github.com/sequenceiq/cloudbreak-rest-client/pull/38#discussion_r23948989>
> :
>
> > @@ -130,6 +133,19 @@ class CloudbreakClient {
> >          return response?.data?.id
> >      }
> >
> > +    def String postRecipe(String recipe, Boolean publicInAccount) throws Exception {
> > +        log.debug("Posting recipe to account...")
> > +        def binding = ["RECIPE": recipe]
> > +        def response;
> > +        if (publicInAccount) {
> > +            response = processPost(Resource.ACCOUNT_RECIPES, binding)
> > +        } else {
> > +            response = processPost(Resource.USER_RECIPES, binding)
> > +        }
> > +        log.debug("Got response: {}", response.data.id)
> > +        return response?.data?.id
>
> Shouldn't the return variable be long? Besides that it looks good.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sequenceiq/cloudbreak-rest-client/pull/38/files#r23948989>
> .
>